src/components/select/filterable-select/filterable-select.pw.tsx

```diff
@@ -0,0 +1,1867 @@
+  FilterableSelectWithActionButtonComponent,
+  FilterableSelectOnChangeEventComponent,
+  FilterableSelectListActionEventComponent,
+  FilterableSelectWithManyOptionsAndVirtualScrolling,
+  FilterableSelectNestedInDialog,
+  SelectionConfirmed,
+} from "../../../../src/components/select/filterable-select/components.test-pw";
+import {
+  commonDataElementInputPreview,
+  getDataElementByValue,
+  helpIcon,
+  tooltipPreview,
+} from "../../../../playwright/components";
+import { alertDialogPreview } from "../../../../playwright/components/dialog";
+import { loader } from "../../../../playwright/components/loader";
+import {
+  boldedAndUnderlinedValue,
+  dropdownButton,
+  filterableSelectAddElementButton,
+  filterableSelectAddNewButton,
+  filterableSelectButtonIcon,
+  multiColumnsSelectListBody,
+  multiColumnsSelectListHeader,
+  multiColumnsSelectListHeaderColumn,
+  multiColumnsSelectListNoResultsMessage,
+  multiColumnsSelectListRow,
+  selectDataComponent,
+  selectElementInput,
+  selectInput,
+  selectList,
+  selectListPosition,
+  selectListWrapper,
+  selectOption,
+  selectOptionByText,
+  selectResetButton,
+} from "../../../../playwright/components/select";
+import { HooksConfig } from "../../../../playwright";
+import {
+  assertCssValueIsApproximately,
+  checkAccessibility,
+  positionOfElement,
+  verifyRequiredAsteriskForLabel,
+} from "../../../../playwright/support/helper";
+import { CHARACTERS, SIZE } from "../../../../playwright/support/constants";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const testPropValue = CHARACTERS.STANDARD;
+const addElementText = "Add a New Element";
+const columns = 3;
+const icon = "add";
+const keyToTrigger = ["ArrowDown", "ArrowUp", "Home", "End"] as const;
+const listOption = "Amber";
+
+test.describe("when focused", () => {
+  test("should have the expected styling when the focusRedesignOptOut is false", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FilterableSelectComponent />);
+
+    const inputElement = commonDataElementInputPreview(page);
+    await inputElement.focus();
+    await expect(inputElement.locator("..")).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(inputElement.locator("..")).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+  });
+
+  test("should have the expected styling when the focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(<FilterableSelectComponent />, {
+      hooksConfig: {
+        focusRedesignOptOut: true,
+      },
+    });
+    const inputElement = commonDataElementInputPreview(page);
+    await inputElement.focus();
+    await expect(inputElement.locator("..")).toHaveCSS(
+      "outline",
+      "rgb(255, 188, 25) solid 3px"
+    );
+  });
+});
+
+test.describe("check props for FilterableSelect component", () => {
```

Comment:
comment (non-blocking): the test descriptions could probably be tidied up but I don't mind if it's done in a later ticket. Same comment for each select file etc

```
check props for FilterableSelect component
  should render label using ${labelValue} special characters
```
could be 
```
FilterableSelect component
  should render label using ${labelValue} special characters

Reply:
not sure I agree with "too many describes" - to me they're useful to organise the tests and group similar ones together, provided the functionality they're testing is actually related. If any I'd say we don't have enough - "check props" is very general for a component like this with loads of props and leads to a block that's 1000+ tests long which is pretty hard to navigate and make sense of (eg if you're wondering where best to add a new test).

But rather than do anything controversial, I'll just leave it as it is for now (but address Ed's comment from above about the wording of that specific block, which I think is fair).